lib/types/string/index.js

```diff
@@ -529,20 +529,28 @@ internals.String = class extends Any {
         return obj;
     }
 
-    trim() {
+    trim(enabled) {
 
-        const obj = this._test('trim', undefined, function (value, state, options) {
+        enabled = enabled === undefined ? true : !!enabled;
```

Comment:
Instead of coercing the value here where someone making the call `.trim('yes please')` would enable trim I would expect us to check the data type like we do in other validation methods. For example: `Hoek.assert(typeof base64Options.paddingRequired === 'undefined' || typeof base64Options.paddingRequired === 'boolean',	         Hoek.assert(typeof base64Options.paddingRequired === 'undefined' || typeof base64Options.paddingRequired === 'boolean', 'paddingRequired must be boolean');`